src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultFamilyMemberViewModel.swift

```diff
@@ -0,0 +1,606 @@
+		familyMemberCoronaTestService.moveTestToBin(familyMemberCoronaTest)
+		onTestDeleted()
+	}
+	
+	func evaluateShowing() {
+		familyMemberCoronaTestService.evaluateShowing(of: familyMemberCoronaTest)
+	}
+	
+	func updateTestResultIfPossible() {
+		guard familyMemberCoronaTest.testResult == .pending else {
+			Log.info("Not refreshing test because status is pending")
+			return
+		}
+		refreshTest()
+	}
+	
+	// MARK: - Private
+	
+	private var familyMemberCoronaTestService: FamilyMemberCoronaTestServiceProviding
+	private var familyMemberCoronaTest: FamilyMemberCoronaTest
+
+	private let onTestDeleted: () -> Void
+	private let onTestCertificateCellTap: (HealthCertificate, HealthCertifiedPerson) -> Void
+
+	private var subscriptions = Set<AnyCancellable>()
+	
+	private var primaryButtonIsLoading: Bool = false {
+		didSet {
+			footerViewModelPublisher.value?.setLoadingIndicator(primaryButtonIsLoading, disable: primaryButtonIsLoading, button: .primary)
+			footerViewModelPublisher.value?.setLoadingIndicator(false, disable: primaryButtonIsLoading, button: .secondary)
+		}
+	}
+	
+	private var showSpecialCaseForNegativeAntigenTest: Bool {
+		return familyMemberCoronaTest.type == .antigen && familyMemberCoronaTest.testResult == .negative
+	}
+
+	private func bindToCoronaTestUpdates() {
+		familyMemberCoronaTestService.coronaTests
+			.sink { [weak self] _ in
+				
+				guard let familyMemberCoronaTest = self?.familyMemberCoronaTest, let familyMemberCoronaTest = self?.familyMemberCoronaTestService.upToDateTest(for: familyMemberCoronaTest) else {
+					return
+				}
+
+				self?.updateForCurrentTestResult(coronaTest: familyMemberCoronaTest)
+			}
+			.store(in: &subscriptions)
+	}
+
+	private func updateForCurrentTestResult(coronaTest: FamilyMemberCoronaTest) {
+		self.familyMemberCoronaTest = coronaTest
+
+		primaryButtonIsLoading = coronaTest.isLoading
+		
+		let sections: [DynamicSection]
+		switch coronaTest.testResult {
+		case .positive:
+			sections = positiveTestResultSections
+		case .negative:
+			if let test = familyMemberCoronaTest.antigenTest, showSpecialCaseForNegativeAntigenTest {
+				sections = negativeAntigenTestResultSections(test: test)
+			} else {
+				sections = negativeTestResultSections
+			}
+		case .invalid:
+			sections = invalidTestResultSections
+		case .pending:
+			sections = pendingTestResultSections
+		case .expired:
+			onTestDeleted()
+			return
+		}
+		dynamicTableViewModelPublisher.value = DynamicTableViewModel(sections)
+		
+		footerViewModelPublisher.value = ExposureSubmissionTestResultFamilyMemberViewModel.footerViewModel(coronaTest: coronaTest)
+	}
+	
+	private func refreshTest() {
+		Log.info("Refresh test.")
+
+		familyMemberCoronaTestService.updateTestResult(for: familyMemberCoronaTest, presentNotification: false) { [weak self] result in
+			guard let self = self else { return }
+			
+			switch result {
+			case let .failure(error):
+				self.errorPublisher.value = error
+			case .success:
+				break
```

Comment:
Is it correct, that its nothing done in case of success? If yes, it would be nice to have a comment here, why its not needed.